.github/workflows/main.yml

```
@@ -0,0 +1,83 @@
+name: CI
+
+on: pull_request
+
+jobs:
+  jjb:
+    runs-on: ubuntu-latest
+    name: Jenkins Job Builder
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '2.7'
+      - name: Restore pip cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - name: Install dependencies
+        run: |
+          echo 'jenkins-job-builder' >> requirements.txt
+          echo 'lxml' >> requirements.txt
+          echo 'requests' >> requirements.txt
+          pip install -r requirements.txt
+      - name: Run tests
+        run: |
+          for d in puppet/modules/jenkins_job_builder/files/theforeman.org; do
+            DIR="."
+            ( cd $d && jenkins-jobs -l debug test -r $DIR )
+            ( cd $d && jenkins-jobs test -r $DIR --config-xml -o output && find output -name config.xml -exec python ../jenkins-lint.py --xml {} + && rm -rf output )
+          done
+
+          for d in puppet/modules/jenkins_job_builder/files/centos.org; do
+            DIR="jobs"
+            ( cd $d && jenkins-jobs -l debug test -r $DIR )
+            ( cd $d && jenkins-jobs test -r $DIR --config-xml -o output && find output -name config.xml -exec python ../jenkins-lint.py --xml {} + && rm -rf output )
+          done
+  puppet:
+    runs-on: ubuntu-latest
+    name: Puppet
+    defaults:
+      run:
+        working-directory: puppet
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Setup ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: '2.5'
+      - name: Install dependencies
+        run: |
+          sudo apt-get update && sudo apt-get install -y yajl-tools
+          gem install bundler
+          bundle config path ~/.vendor/bundle
+      - uses: actions/cache@v2
+        with:
+          path: ~/.vendor/bundle
+          key: ${{ runner.os }}-gems-unit-
```

This should include the Ruby version and hashfiles
```suggestion
          key: ${{ runner.os }}-gems-2.5-${{ hashFiles('**/Gemfile.lock') }}
```

---

If there is never a `Gemfile.lock` at this point in the steps, I do not get why I would include hashing it?

---

I think that `bundle install` generates a `Gemfile.lock` and in the post step it will create a bundle with the correct hash.

---

That makes sense. When it runs the next time, since there is no `Gemfile.lock` created yet at this step, it will use `${{ runner.os }}-gems-2.5-` as the lookup key right? So the hash of the `Gemfile.lock` seems to be pointless without creating the lockfile in the step above it.